test/test_mcmc.py

```diff
@@ -16,24 +16,27 @@
 import numpyro
 import numpyro.distributions as dist
 from numpyro.distributions import constraints
-from numpyro.infer import HMC, MCMC, NUTS
-from numpyro.infer.mcmc import hmc
+from numpyro.infer import HMC, MCMC, NUTS, SA
+from numpyro.infer.mcmc import hmc, _get_proposal_loc_and_scale
 from numpyro.infer.util import initialize_model
 from numpyro.util import fori_collect
 
 
-@pytest.mark.parametrize('kernel_cls', [HMC, NUTS])
+@pytest.mark.parametrize('kernel_cls', [HMC, NUTS, SA])
 @pytest.mark.parametrize('dense_mass', [False, True])
 def test_unnormalized_normal_x64(kernel_cls, dense_mass):
     true_mean, true_std = 1., 0.5
-    warmup_steps, num_samples = 1000, 8000
+    warmup_steps, num_samples = (100000, 100000) if kernel_cls is SA else (1000, 8000)
```

Comment:
how long do these new tests take?

Reply:
They are super fast (a few seconds for all of them as long as we disable progbar). A few observations are:
+ n_eff/num_samples is small (which indicates that the samples are correlated).
+ n_eff/s seems to be better than NUTS.

So with the same num_samples, this will not be as good as NUTS. But given the same amount of time, this can be better than NUTS.